src/d3a/models/area/__init__.py

```diff
@@ -404,20 +404,29 @@ def tick(self):
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
-                if is_external_matching_enabled():
-                    bid_offer_matcher.match_algorithm.update_area_uuid_markets_mapping(
-                        {self.uuid: self.all_markets})
-                else:
-                    for market in self.all_markets:
-                        bid_offer_pairs = bid_offer_matcher.calculate_recommendation(
-                            *market.open_bids_and_offers, self.now)
-                        while bid_offer_pairs:
-                            bid_offer_pairs = bid_offer_matcher.calculate_recommendation(
-                                *market.open_bids_and_offers, self.now)
-                            market.match_recommendation(bid_offer_pairs)
+                self._match_bids_offers()
 
         self.events.update_events(self.now)
 
+    def _match_bids_offers(self) -> None:
+        """Match bids and offers for all markets."""
+        if is_external_matching_enabled():
+            # Update the open offer bids cache that the myco client will request
+            bid_offer_matcher.match_algorithm.update_area_uuid_markets_mapping(
+                {self.uuid: self.all_markets})
+            return
+        # If the external matching is not enabled, get and match bid offer pairs
+        for market in self.all_markets:
+            bid_offer_pairs = True
```

Comment:
Nit: this one seems a bit counterintuitive to me, because the content of `bid_offer_pairs` shouldn't generally be a boolean. I understand this is a sort of trick to get the `while` loop started, but in this case I would prefer having a more verbose `while True` loop. For example:
```python
while True:
    bid_offer_pairs = ...
    if not bid_offer_pairs:
        break
```

Reply:
you are right, the do while loop is missed in python indeed.
Thanks!